hack/generator/spec/swagger-resource-references-onepager.md

```diff
@@ -0,0 +1,54 @@
+
+Not every team does references the same way (which is fine), but that means that as a client of the Swagger it's impossible to tell if a given field is a reference or not.
+
+## Digging in
+The easiest way to accomplish this is just to expose "is this field a reference" as an extension in Swagger using `format`. Probably something like: 
+```
+    "SubResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "resourceId",
+          "description": "Resource ID."
+        }
+      },
+      "description": "Reference to another subresource."
+    },
+```
+
+## What is Azure Service Operator (ASO)?
+It's a Kubernetes operator which supports creating Azure resources. You can read more [here](https://github.com/Azure/azure-service-operator). It allows customers to create Azure resources directly in their Kubernetes cluster by defining YAML files representing the Azure resource and then doing `kubectl apply -f <my-yaml>`.
+
+## How do resource references work in ASO?
+ASO allows you to represent various Azure resources in Kubernetes. As a consequence of the fact that the resource is reflected in Kubernetes, you also can _refer_ to resources in one of two ways:
+1. A reference to the resource via its ARM ID, the same as you would do in an ARM template or the SDK.
+2. A reference to the resource via its Kubernetes name. This is more natural to Kubernetes developers and plays to Kubernetes strengths by allowing you to more easily apply the same specification in multiple places (generating distinct resources with distinct ARM IDs) without having to update the resource specification for each kind of deployment. An example of this would be creating the same resource in Kubernetes in two different namesapces which map to two different subscriptions in Azure - the "Kubernetes reference" looks the same but the ARM ID of the actual referenced resource is different.
+
+You can read more about the references design [here](https://github.com/Azure/k8s-infra/blob/master/hack/generator/spec/type_references_and_ownership.md)
+
+## Why do we want this ability?
+In a lot of ways, the maintainers of ASO (that's my team and I) are in the same position as the SDK team: there are tons of Azure services and there's no way we can keep up with the pace of new API versions/services without some sort of automation helping us. We are currently most of the way through developing a code-generator that generates Kubernetes CRDs from the azure-rest-api-specs Swager documents - you can see this generator here: [k8sinfra](https://github.com/Azure/k8s-infra).
+
+In order for the generator to generate the right shape for resources that have references, we need a way to determine that a particular field is a reference, so that we can replace the simple `string` field with something like:
```

Comment:
```suggestion
In order for the generator to emit the right shape for resources that have references, we need a reliable/authoritative way to determine that a particular field is a reference, so that we can replace the simple `string` field with something like:
```

Reply:
Done